lib/utils.js

```diff
@@ -132,7 +132,9 @@ exports.compact = function (obj, references) {
         var compacted = [];
 
         for (var i = 0; i < obj.length; ++i) {
-            if (typeof obj[i] !== 'undefined') {
+            if (Array.isArray(obj[i]) || Object.prototype.toString.call(obj[i]) === '[object Object]') {
```

Comment:
There is a problem with `null` which type is also an `object`. We could add a check for `null` though.